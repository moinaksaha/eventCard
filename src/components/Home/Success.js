import React, { Component } from 'react';

import { Button, Glyphicon } from 'react-bootstrap';

import styles from '../../containers/Home/Home.css'

export default class Success extends Component{

	constructor(){
    	super();
    }
      
    resetForm = () => {
        console.log("happy resetting form");
        const { resetFormData } = this.props;
        resetFormData();
    }

	render = () => {

		return (

                <div className={`${styles.successContainer} text-center center-block`}>

                    <div className={`${styles.heading}`}>

                        <Glyphicon glyph={`ok`}
								   className={`${styles.inputValidationIcon}`}/>

                        <div>Form Submitted Successfully</div>

                        
                    
                    </div>

                    <div className={`${styles.content} text-center`}>
                    
                        <Button type="button" 
                                className={`${styles.successButton}`}
                                onClick={this.resetForm}>

                            Re Submit Form

                        </Button>

                        <div>*Alternatively, you can refresh the page.</div>
                        <div>*We're not saving any data here ;)</div>
                    
                    </div>

                </div>
		);
	}
};

Success.defaultProps = {

};
