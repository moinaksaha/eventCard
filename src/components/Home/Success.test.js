import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

import Success from './Success';

import styles from '../../containers/Home/Home.css';

const mockRemoveGreeting = jest.fn();

// describe what we are testing
describe('Form Submitted Successfully Component', () => {
 
    // make our assertion and what we expect to happen 
    it('should render without throwing an error', () => {

      expect(mount(<Success />).exists(<div className={`${styles.successContainer} text-center center-block`}></div>)).toBe(true)

    });

    it('calls the passed in resetFormData function when button is clicked', () => {

        mount(<Success resetFormData={mockRemoveGreeting}/>).find('button').simulate('click');

        expect(mockRemoveGreeting).toBeCalled();

    });

    it('renders correctly', () => {
        const successSnap = renderer.create(
          <Success />
        ).toJSON();
        expect(successSnap).toMatchSnapshot();
    });
})