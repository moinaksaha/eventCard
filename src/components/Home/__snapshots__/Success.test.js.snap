// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Submitted Successfully Component renders correctly 1`] = `
<div
  className="undefined text-center center-block"
>
  <div
    className="undefined"
  >
    <span
      className="undefined glyphicon glyphicon-ok"
    />
    <div>
      Form Submitted Successfully
    </div>
  </div>
  <div
    className="undefined text-center"
  >
    <button
      className="undefined btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Re Submit Form
    </button>
    <div>
      *Alternatively, you can refresh the page.
    </div>
    <div>
      *We're not saving any data here ;)
    </div>
  </div>
</div>
`;
